crates/bevy_ecs/src/bundle.rs

```diff
@@ -224,9 +225,9 @@ macro_rules! tuple_impl {
             where
                 F: FnMut(&mut T) -> OwningPtr<'_>
             {
-                // Rust guarantees that tuple calls are evaluated 'left to right'.
+                // SAFETY: Rust guarantees that tuple calls are evaluated 'left to right'.
                 // https://doc.rust-lang.org/reference/expressions.html#evaluation-order-of-operands
-                ($(<$name as Bundle>::from_components(ctx, func),)*)
+                ($(unsafe { <$name as Bundle>::from_components(ctx, func) },)*)
```

Comment:
Moving the unsafe block outside the macro expansion triggers the [unused-unsafe](https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#unused-unsafe) lint.

Reply:
Perhaps it would be useful to compare compile times with and without this change? If it is faster to move the `unsafe` block out of the expansion, then I think it's ok to `#[allow(...)]` it.